KomaMRICore/src/simulation/SimulatorCore.jl

```diff
@@ -222,24 +232,11 @@ function simulate(
     sig .*= get_adc_phase_compensation(seq)
     Xt = Xt |> cpu
     if sim_params["gpu"] GC.gc(true); CUDA.reclaim() end
-    # Output
-    if sim_params["return_type"] == "state"
-        out = Xt
-    elseif sim_params["return_type"] == "mat"
-        out = sig
-    elseif sim_params["return_type"] == "raw"
-        # To visually check the simulation blocks
-        sim_params_raw = copy(sim_params)
-        sim_params_raw["sim_method"] = string(sim_params["sim_method"])
-        sim_params_raw["gpu"] = sim_params["gpu"]
-        sim_params_raw["Nthreads"] = sim_params["Nthreads"]
-        sim_params_raw["t_sim_parts"] = t_sim_parts
-        sim_params_raw["type_sim_parts"] = excitation_bool
-        sim_params_raw["Nblocks"] = length(parts)
-        sim_params_raw["sim_time_sec"] = timed_tuple.time
-        out = simulation_output(sig, seq, obj.name, sys, sim_params_raw)
-        #out = signal_to_raw_data(sig, seq; phantom_name=obj.name, sys=sys, sim_params=sim_params_raw)
-    end
+    out = simulation_output(
+        sim_params["return_type"];
+        Xt, sig, seq, obj, sys, sim_params, t_sim_parts, excitation_bool,
+        Nparts=length(parts), timed_tuple_time=timed_tuple.time
+    )
```

Comment:
This is weird, if you do `simulation_output(MagState)` it would throw an error. I think all the additional stuff should be in sim_params.

Reply:
Now we rolled back. Addressed in [dfda824](https://github.com/cncastillo/KomaMRI.jl/commit/dfda8249dd5d5d993ee0f6a27eb50f1fcdfdb4ee)